kvserver: TestStoreRangeReadoptedLHSFollower is flaky

See https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/8665359?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true


```
Failed
=== RUN   TestStoreRangeReadoptedLHSFollower/withMerge=true
    client_merge_test.go:3162: could not remove [n3,s3] replicas from range containing /Table/Max: AdminChangeReplicas error: cannot remove learner while snapshot is in flight
        (1) AdminChangeReplicas error: cannot remove learner while snapshot is in flight
          | -- cause hidden behind barrier
          | AdminChangeReplicas error: cannot remove learner while snapshot is in flight
          | (1) attached stack trace
          |   -- stack trace:
          |   | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).changeReplicas
          |   |   github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:762
          |   | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RemoveVoters
          |   |   github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:918
          |   | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RemoveVotersOrFatal
          |   |   github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:926
          |   | github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestStoreRangeReadoptedLHSFollower.func1
          |   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/client_merge_test.go:3162
          |   | github.com/cockroachdb/cockroach/pkg/testutils.RunTrueAndFalse.func1
          |   |   github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:23
          |   | testing.tRunner
          |   |   GOROOT/src/testing/testing.go:1446
          |   | runtime.goexit
          |   |   GOROOT/src/runtime/asm_amd64.s:1594
          | Wraps: (2) AdminChangeReplicas error
          | Wraps: (3) forced error mark
          |   | "replication change in progress"
          |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
          | Wraps: (4)
          |   | (opaque error wrapper)
          |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          |   | reportable 0:
          |   |
          |   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.init
          |   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_command.go:1317
          |   | runtime.doInit
          |   |   GOROOT/src/runtime/proc.go:6348
          |   | runtime.doInit
          |   |   GOROOT/src/runtime/proc.go:6325
          |   | runtime.main
          |   |   GOROOT/src/runtime/proc.go:233
          |   | runtime.goexit
          |   |   GOROOT/src/runtime/asm_amd64.s:1594
          | Wraps: (5) cannot remove learner while snapshot is in flight
          | Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *markers.withMark (4) *errbase.opaqueWrapper (5) *errutil.leafError
        Error types: (1) *barriers.barrierErr
    --- FAIL: TestStoreRangeReadoptedLHSFollower/withMerge=true (3.24s)
```

Jira issue: CRDB-24793

Flakes for me as well:
https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/8826943?showRootCauses=true&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true
```
Failed
=== RUN   TestStoreRangeReadoptedLHSFollower/withMerge=false
    client_merge_test.go:3162: could not remove [n3,s3] replicas from range containing /Table/Max: AdminChangeReplicas error: cannot remove learner while snapshot is in flight
        (1) AdminChangeReplicas error: cannot remove learner while snapshot is in flight
          | -- cause hidden behind barrier
          | AdminChangeReplicas error: cannot remove learner while snapshot is in flight
          | (1) attached stack trace
          |   -- stack trace:
          |   | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).changeReplicas
          |   |   github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:762
          |   | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RemoveVoters
          |   |   github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:918
          |   | github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).RemoveVotersOrFatal
          |   |   github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:926
          |   | github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestStoreRangeReadoptedLHSFollower.func1
          |   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/client_merge_test.go:3162
          |   | github.com/cockroachdb/cockroach/pkg/testutils.RunTrueAndFalse.func1
          |   |   github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:23
          |   | testing.tRunner
          |   |   GOROOT/src/testing/testing.go:1446
          |   | runtime.goexit
          |   |   GOROOT/src/runtime/asm_amd64.s:1594
          | Wraps: (2) AdminChangeReplicas error
          | Wraps: (3) forced error mark
          |   | "replication change in progress"
          |   | github.com/cockroachdb/errors/withstack/*withstack.withStack::
          | Wraps: (4)
          |   | (opaque error wrapper)
          |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
          |   | reportable 0:
          |   |
          |   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.init
          |   |   github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_command.go:1311
          |   | runtime.doInit
          |   |   GOROOT/src/runtime/proc.go:6348
          |   | runtime.doInit
          |   |   GOROOT/src/runtime/proc.go:6325
          |   | runtime.main
          |   |   GOROOT/src/runtime/proc.go:233
          |   | runtime.goexit
          |   |   GOROOT/src/runtime/asm_amd64.s:1594
          | Wraps: (5) cannot remove learner while snapshot is in flight
          | Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *markers.withMark (4) *errbase.opaqueWrapper (5) *errutil.leafError
        Error types: (1) *barriers.barrierErr
    --- FAIL: TestStoreRangeReadoptedLHSFollower/withMerge=false (1.44s)
```

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -3160,7 +3160,13 @@ func TestStoreRangeReadoptedLHSFollower(t *testing.T) {
 			}
 			return nil
 		})
-		tc.RemoveVotersOrFatal(t, lhsDesc.StartKey.AsRawKey(), tc.Target(2))
+		// Need to retry because removing voters could fail while snapshots are
+		// still in flight and cause "cannot remove learner while snapshot is in
+		// flight" error.
+		testutils.SucceedsSoon(t, func() error {
+			_, err := tc.RemoveVoters(lhsDesc.StartKey.AsRawKey(), tc.Target(2))
+			return err
+		})
 
 		if withMerge {
 			// Merge the two ranges together.

